contracts/extensions/EIP3009.sol

```diff
@@ -55,30 +63,19 @@ abstract contract EIP3009 is EIP712Domain {
         bytes32 r,
         bytes32 s
     ) external {
-        if (block.timestamp < validAfter) revert EIP3009_AuthorizationNotYetValid();
-        if (block.timestamp > validBefore) revert EIP3009_AuthorizationExpired();
-        if (_authorizationStates[from][nonce]) revert EIP3009_AuthorizationUsed();
-        /** @dev Prevent front-running: only 'to' address can execute this */
-        if (msg.sender != from) revert EIP3009_UnauthorizedCaller();
-
-        /** @dev Prevent reentrancy attack */
-        _authorizationStates[from][nonce] = true;
-
-        /** @dev Check signature */
-        bytes memory data = abi.encode(
+        _transferWithAuthorization(
             TRANSFER_WITH_AUTHORIZATION_TYPEHASH,
             from,
             to,
             value,
             validAfter,
             validBefore,
-            nonce
+            nonce,
+            v,
+            r,
+            s
         );
 
-        if (EIP712.recover(DOMAIN_SEPARATOR, v, r, s, data) != from) revert EIP3009_InvalidSignature();
-
-        emit AuthorizationUsed(from, nonce);
-
         _burnTokens(from, value);
```

Comment:
This method is not about minting/buringing.
Here we should be calling `_transfer` function from ERC20 base. So just moving tokens from whoever authorized moving them to whenever they were authorized